File: app/src/main/java/com/montreal/wtm/model/Session.java
package com.montreal.wtm.model;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;
import java.util.ArrayList;

@IgnoreExtraProperties
public class Session implements Parcelable {

    @PropertyName("id")
    public int id;

    @PropertyName("complexity")
    public String complexity;

    @PropertyName("title")
    public String title;

    @PropertyName("description")
    public String description;

    @PropertyName("language")
    public String language;

    @PropertyName("speakers")
    public ArrayList<Integer> speakersId;

    @PropertyName("tags")
    public ArrayList<String> tags;

    public Session(int id, String complexity, String title, String description, String language,
        ArrayList<Integer> speakersId, ArrayList<String> tags) {
        this.id = id;
        this.complexity = complexity;
        this.title = title;
        this.description = description;
        this.language = language;
        this.speakersId = speakersId;
        this.tags = tags;
    }

    public int getId() {
        return id;
    }

    public String getComplexity() {
        return complexity;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getLanguage() {
        return language;
    }

    public ArrayList<Integer> getSpeakersId() {
        return speakersId;
    }

    public ArrayList<String> getTags() {
        return tags;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.id);
        dest.writeString(this.complexity);
        dest.writeString(this.title);
        dest.writeString(this.description);
        dest.writeString(this.language);
        dest.writeSerializable(this.speakersId);
        dest.writeSerializable(this.tags);
    }

    public Session() {
    }

    protected Session(Parcel in) {
        this.id = in.readInt();
        this.complexity = in.readString();
        this.title = in.readString();
        this.description = in.readString();
        this.language = in.readString();
        this.speakersId = (ArrayList<Integer>) in.readSerializable();
        this.tags = (ArrayList<String>) in.readSerializable();
    }

    public static final Creator<Session> CREATOR = new Creator<Session>() {
        @Override
        public Session createFromParcel(Parcel source) {
            return new Session(source);
        }

        @Override
        public Session[] newArray(int size) {
            return new Session[size];
        }
    };
}

File: app/src/main/java/com/montreal/wtm/model/Session.kt
package com.montreal.wtm.model

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.database.IgnoreExtraProperties
import com.google.firebase.database.PropertyName
import java.util.ArrayList

@IgnoreExtraProperties
class Session : Parcelable {
  val TECHNICAL_SKILL = "Technical Skill"

  @PropertyName("id")
  var id: Int = 0

  @PropertyName("complexity")
  var complexity: String = ""

  @PropertyName("title")
  lateinit var title: String

  @PropertyName("description")
  lateinit var description: String

  @PropertyName("language")
  var language: String = "English"

  @PropertyName("roomId")
  var roomId: Int = 0

  @PropertyName("type")
  var type: String? = null

  @PropertyName("speakers")
  var speakers: ArrayList<Int> = ArrayList<Int>()

  @PropertyName("tags")
  var tags: ArrayList<String> = ArrayList<String>()

  constructor()

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeInt(this.id)
    dest.writeString(this.complexity)
    dest.writeString(this.title)
    dest.writeString(this.description)
    dest.writeString(this.language)
    dest.writeSerializable(if (this.speakers == null) ArrayList<Int>() else this.speakers)
    dest.writeSerializable(this.tags)
  }

  protected constructor(`in`: Parcel) {
    this.id = `in`.readInt()
    this.complexity = `in`.readString() ?: ""
    this.title = `in`.readString() ?: ""
    this.description = `in`.readString() ?: ""
    this.language = `in`.readString() ?: ""
    this.speakers = `in`.readSerializable() as ArrayList<Int>
    this.tags = `in`.readSerializable() as ArrayList<String>
  }

  companion object {

    @JvmField
    val CREATOR: Parcelable.Creator<Session> = object : Parcelable.Creator<Session> {
      override fun createFromParcel(source: Parcel): Session {
        return Session(source)
      }

      override fun newArray(size: Int): Array<Session?> {
        return arrayOfNulls(size)
      }
    }
  }

  fun isTechnicalTalk():Boolean {
    for(tag in tags) {
      if(tag.contains(TECHNICAL_SKILL, true)) {
        return true
      }
    }
    return false
  }

  fun isIncludedIn(savedSessions:HashMap<String, Boolean>? ) :Boolean {
    return savedSessions?.get(id.toString())?:false;
  }
}
